### DISCUSSION TITLE
Websocket - wss support.

### DISCUSSION THREAD

**Discussion by JadKHaddad-2** [poem-web/poem#400] [Q&A]
I built a webapp  using  poem. Websocket connections work fine as log as I use "ws:{ hostname }". In production my app  runs behind a reverse proxy that uses https, so that the Websocket connections are established using "wss:{ hostname }" and they fail. Do I  have  to make sure that poem supports wss or is it a reverse proxy problem that I can't fix?

**Comment by AlexMikhalev** [comment]
wss requires security certificates to be available on the server. You need to pass certificates to poem server or SSL terminate on the proxy. If your SSL is on the proxy like nginx, then redirect to ws endpoint inside server or location parameters inside conf.

**Reply by JadKHaddad-2** [reply]
I dont have access to the certificates

**Reply by AlexMikhalev** [reply]
Then " it a reverse proxy problem that I can't fix?" Ask whoever maintains your reverse proxy to set certificates for a secure websocket. It may be the case of self-signed certificates failing for WSS.
